cmd/collectors/zapi/plugins/volume/volume.go

```diff
@@ -20,84 +20,171 @@ type Volume struct {
+	data := dataMap[v.Object]
+	if v.currentVal >= v.PluginInvocationRate {
+		v.currentVal = 0
 
 		// invoke disk-encrypt-get-iter zapi and populate disk info
-		disks, err1 := my.getEncryptedDisks()
+		disks, err1 := v.getEncryptedDisks()
 		// invoke aggr-status-get-iter zapi and populate aggr disk mapping info
-		aggrDiskMap, err2 := my.getAggrDiskMapping()
+		aggrDiskMap, err2 := v.getAggrDiskMapping()
 
 		if err1 != nil {
 			if errors.Is(err1, errs.ErrNoInstance) {
-				my.Logger.Debug().Err(err1).Msg("Failed to collect disk data")
+				v.Logger.Debug().Err(err1).Msg("Failed to collect disk data")
 			} else {
-				my.Logger.Error().Err(err1).Msg("Failed to collect disk data")
+				v.Logger.Error().Err(err1).Msg("Failed to collect disk data")
 			}
 		}
 		if err2 != nil {
 			if errors.Is(err2, errs.ErrNoInstance) {
-				my.Logger.Debug().Err(err2).Msg("Failed to collect aggregate-disk mapping data")
+				v.Logger.Debug().Err(err2).Msg("Failed to collect aggregate-disk mapping data")
 			} else {
-				my.Logger.Error().Err(err2).Msg("Failed to collect aggregate-disk mapping data")
+				v.Logger.Error().Err(err2).Msg("Failed to collect aggregate-disk mapping data")
 			}
 		}
 		// update aggrsMap based on disk data and addr disk mapping
-		my.updateAggrMap(disks, aggrDiskMap)
+		v.updateAggrMap(disks, aggrDiskMap)
+	}
+
+	volumeCloneMap, err := v.getVolumeCloneInfo()
+
+	if err != nil {
+		v.Logger.Error().Err(err).Msg("Failed to update clone data")
 	}
 
 	// update volume instance labels
-	my.updateVolumeLabels(data)
+	v.updateVolumeLabels(data, volumeCloneMap)
 
-	my.currentVal++
+	v.currentVal++
 	return nil, nil
 }
 
-func (my *Volume) updateVolumeLabels(data *matrix.Matrix) {
+func (v *Volume) updateVolumeLabels(data *matrix.Matrix, volumeCloneMap map[string]volumeClone) {
+	var err error
 	for _, volume := range data.GetInstances() {
 		aggrUUID := volume.GetLabel("aggrUuid")
-		_, exist := my.aggrsMap[aggrUUID]
+		_, exist := v.aggrsMap[aggrUUID]
 		volume.SetLabel("isHardwareEncrypted", strconv.FormatBool(exist))
+
+		name := volume.GetLabel("volume")
+		svm := volume.GetLabel("svm")
+		key := name + svm
+
+		if vc, ok := volumeCloneMap[key]; ok {
+			volume.SetLabel("clone_parent_snapshot", vc.parentSnapshot)
+			volume.SetLabel("clone_parent_volume", vc.parentVolume)
+			volume.SetLabel("clone_parent_svm", vc.parentSvm)
+			splitEstimate := data.GetMetric("clone_split_estimate")
+			if splitEstimate == nil {
+				if splitEstimate, err = data.NewMetricFloat64("clone_split_estimate"); err != nil {
+					v.Logger.Error().Err(err).Str("metric", "clone_split_estimate").Msg("add metric")
+					continue
+				}
+			}
+
+			// splitEstimate is 4KB blocks, Convert to bytes as in REST
+
+			var splitEstimateBytes float64
+			if splitEstimateBytes, err = strconv.ParseFloat(vc.splitEstimate, 64); err != nil {
+				v.Logger.Error().Err(err).Str("metric", "clone_split_estimate").Str("value", vc.splitEstimate).Msg("parse metric")
```

Comment:
minor: since the metric is static this could be rewritten as
`v.Logger.Error().Err(err).Str("clone_split_estimate", vc.splitEstimate).Msg("parse clone_split_estimate")`